src/archive/zip.rs

```diff
@@ -261,14 +261,15 @@ fn get_last_modified_time(file: &fs::File) -> DateTime {
 }
 
 fn set_last_modified_time(zip_file: &ZipFile, path: &Path) -> crate::Result<()> {
-    let modification_time_in_seconds = zip_file
-        .last_modified()
-        .to_time()
-        .expect("Zip archive contains a file with broken 'last modified time'")
-        .unix_timestamp();
+    let modification_time = zip_file.last_modified().to_time();
+
+    let time_in_seconds = match modification_time {
```

Comment:
btw you can use `let else` here

Reply:
First I did it with `let-else`, but it's that use-case where I start questioning my rustfmt gut. 

```rust
let modification_time = zip_file
    .last_modified()
    .to_time()
else {
    return Ok(());
};
```

:eyes: do you like this?

Reply:
I think rustfmt just gave up since it is not implemented, and you can manually format it however you want

You can separate it into 2 statements like what you have right now if that makes it easier to format manually

Reply:
Oh, I see what you mean, will do and merge.